client/http/client.go

```diff
@@ -271,7 +275,7 @@ func (c *client) request(
 		zap.String("caller-id", c.callerID),
 	}
 	log.Debug("[pd] request the http url", logFields...)
-	req, err := http.NewRequestWithContext(ctx, method, url, body)
+	req, err := http.NewRequestWithContext(ctx, method, url, bytes.NewBuffer(body))
```

Comment:
If `body` is nil, is it OK to pass it into the `bytes.NewBuffer`? Should we add a check to use `http.NoBody` like before?

Reply:
nil is OK, IDE shows other packages uses `bytes.NewBuffer(nil)` safely.